pkg/provisioner/ironic/testserver/server.go

```diff
@@ -0,0 +1,104 @@
+package testserver
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+// New returns a MockServer
+func New(t *testing.T, name string) *MockServer {
+	mux := http.NewServeMux()
+	return &MockServer{
+		t:    t,
+		name: name,
+		mux:  mux,
+	}
+}
+
+// MockServer is a simple http testing server
+type MockServer struct {
+	t            *testing.T
+	mux          *http.ServeMux
+	name         string
+	Requests     string
+	FullRequests []*http.Request
```

Comment:
Is this field used somewhere?

Reply:
Not yet. I'm anticipating wanting it when we have tests that try to PUT or POST data, so we can see what that data is. I could remove it for now, but it's likely to come back in some form in the next PR anyway.

Reply:
That's ok then